endpoints/api/namespacequota.py

```diff
@@ -75,252 +74,322 @@ class OrganizationQuota(ApiResource):
+            "required": ["type", "threshold_percent"],
             "properties": {
-                "percent_of_limit": {
-                    "type": "integer",
-                    "description": "Percentage of quota at which to do something",
+                "type": {
+                    "type": "string",
+                    "description": 'Type of quota limit: "Warning" or "Reject"',
                 },
-                "quota_type_id": {
+                "threshold_percent": {
                     "type": "integer",
-                    "description": "Quota type Id",
+                    "description": "Quota threshold, in percent of quota",
                 },
             },
         },
     }
 
-    @nickname("getOrganizationQuotaLimit")
-    def get(self, namespace):
-        orgperm = OrganizationMemberPermission(namespace)
-        userperm = UserReadPermission(namespace)
-
-        if not orgperm.can() and not userperm.can():
+    @nickname("listOrganizationQuotaLimit")
+    def get(self, orgname, quota_id):
+        orgperm = OrganizationMemberPermission(orgname)
+        if not orgperm.can():
             raise Unauthorized()
 
-        quota_limits = list(model.namespacequota.get_namespace_limits(namespace))
-
-        return {"quota_limits": [quota_limit_view(namespace, limit) for limit in quota_limits]}, 200
+        org = model.organization.get_organization(orgname)
+        quota = get_quota(org, quota_id)
+        return [
+            limit_view(limit)
+            for limit in model.namespacequota.get_namespace_quota_limit_list(quota)
+        ]
 
     @nickname("createOrganizationQuotaLimit")
     @validate_json_request("NewOrgQuotaLimit")
-    def post(self, namespace):
-        """
-        Create a new organization quota.
-        """
-
-        orgperm = AdministerOrganizationPermission(namespace)
-        superperm = SuperUserPermission()
-
-        if not superperm.can():
-            if orgperm.can():
-                if config.app_config.get("DEFAULT_SYSTEM_REJECT_QUOTA_BYTES") != 0:
-                    raise Unauthorized()
-            else:
+    def post(self, orgname, quota_id):
+        orgperm = AdministerOrganizationPermission(orgname)
+
+        if not features.SUPER_USERS or not SuperUserPermission().can():
+            if (
+                not orgperm.can()
+                or not config.app_config.get("DEFAULT_SYSTEM_REJECT_QUOTA_BYTES") != 0
+            ):
                 raise Unauthorized()
 
         quota_limit_data = request.get_json()
-        quota = model.namespacequota.get_namespace_limit(
-            namespace, quota_limit_data["quota_type_id"], quota_limit_data["percent_of_limit"]
+        quota_type = quota_limit_data["type"]
+        quota_limit_threshold = quota_limit_data["threshold_percent"]
+
+        org = model.organization.get_organization(orgname)
+        quota = get_quota(org, quota_id)
+
+        quota_limit = model.namespacequota.get_namespace_quota_limit_list(
```

Comment:
This will not return a list as there should only be 1 limit which a set threshold limit. Return type will be list according to the function name.